src/features/Search/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx

```diff
@@ -1,153 +1,194 @@
-          />
-        </HStack>
-
-        <HStack width="full" gap={6}>
-          <RangeSlider
-            control={control}
-            name="lengthRange"
-            label="Length"
-            min={0}
-            max={50}
-          />
-          <RangeSlider
-            control={control}
-            name="releaseYearRange"
-            label="Release date"
-            min={1970}
-            max={currentYear}
-          />
-        </HStack>
-      </VStack>
-      <Button type="submit">Submit</Button>
+    <form>
+      <HStack width="full" alignItems="end" gap={4}>
+        <TextInput
+          control={control}
+          name="title"
+          label="Search"
+          leftElement={<Icon name="carbon:search" />}
+        />
+
+        <MultiSelect
+          control={control}
+          name="languages"
+          label="Language"
+          options={languageOptions}
+          placeholder="Any"
+          displayLimit={1}
+          closeMenuOnSelect={false}
+        />
+
+        <MultiSelect
+          control={control}
+          name="platforms"
+          label="Platform"
+          options={platformOptions}
+          placeholder="Any"
+          displayLimit={1}
+          closeMenuOnSelect={false}
+        />
+
+        <IconButton
+          aria-label="Open modal with more filter options"
+          onClick={onOpen}
+          icon={<Icon name="carbon:settings-adjust" />}
+          colorScheme="gray"
+          size="sm"
+        />
+      </HStack>
+      <Modal
+        size="5xl"
+        onClose={onClose}
+        isOpen={isOpen}
+        preserveScrollBarGap
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalBody padding={12}>
+            <VStack gap={14}>
+              <HStack width="full">
+                <TextInput
+                  control={control}
+                  name="title"
+                  leftElement={<Icon name="carbon:search" />}
+                />
+              </HStack>
+
+              <HStack width="full">
+                <MultiSelect
+                  control={control}
+                  name="languages"
+                  label="Language"
+                  options={languageOptions}
+                  placeholder="Any"
+                  displayLimit={1}
+                  closeMenuOnSelect={false}
+                />
+
+                <MultiSelect
+                  control={control}
+                  name="originalLanguages"
+                  label="Original language"
+                  options={languageOptions}
+                  placeholder="Any"
+                  displayLimit={1}
+                  closeMenuOnSelect={false}
+                />
+
+                <MultiSelect
+                  control={control}
+                  name="platforms"
+                  label="Platform"
+                  options={platformOptions}
+                  placeholder="Any"
+                  displayLimit={1}
+                  closeMenuOnSelect={false}
```

Comment:
You can use object for default `multiselect` options  like `placeholder`, `displayLimit` and others to reduce code duplication